Update Reader bundle id and add keychain access group

mokagio opened the pull request:
Part of https://linear.app/a8c/issue/CMM-260/configure-reader-target. See also paaHJt-8hM-p2.

mokagio committed:
Move Reader bundle identifier definition to `xcconfig`

mokagio committed:
Update Reader bundle identifier to new one with correct app prefix id

Internal ref paaHJt-8hM-p2

mokagio committed:
Add keychain access group to Reader entitlements

mokagio committed:
Update Reader iTunes ID

See paaHJt-8hM-p2

dangermattic commented:
<!--
  1 Error: PR requires at least one label...
  0 Warnings
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR requires at least one label.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

mokagio committed:
Add required `NSPhotoLibraryUsageDescription` to Reader

It's surprising that 4d7578f667080139236a1ae83ea72402f84167ec did not
include this key while also being successfully distributed to
TestFlight...

mokagio commented:
The failed TestFlight upload has to do with missing privacy descriptions, at least `NSPhotoLibraryUsageDescription`. 

It's surprising that 4d7578f667080139236a1ae83ea72402f84167ec did not    include this key while also being successfully distributed to    TestFlight...

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-iOS/milestone/308">25.9</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

mokagio commented:
https://github.com/wordpress-mobile/WordPress-iOS/pull/24478/commits/576f8d47abd82ecfdcc23e3eb703425be698521a addressed the TestFlight processing issue

<img alt="IMG_808397CE6B66-1" src="https://github.com/user-attachments/assets/bf09c9f1-57e8-4bb4-a3e1-c11c0b75ea29" width=300/>

Just waiting for https://github.com/wordpress-mobile/WordPress-iOS/pull/24471 to land, given it's already in the merge queue, so I can update the base branch.
